flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/filesystem/FileSystemTableFactory.java

```diff
@@ -71,6 +71,12 @@
 			.withDescription("The default partition name in case the dynamic partition" +
 					" column value is null/empty string");
 
+	public static final ConfigOption<Boolean> SINK_SHUFFLE_BY_PARTITION = key("sink.shuffle-by-partition.enable")
+			.booleanType()
+			.defaultValue(false)
+			.withDescription("Before sink, can shuffle by dynamic partition fields to sink parallelisms," +
+					" this can greatly reduce the number of files. But will lead to data skew too.");
+
```

Comment:
How about this ？"The option to enable shuffle data by dynamic partition fields in sink phase, this can greatly reduce the number of file for filesystem sink but may lead data skew, the default value is disabled."